### PR TITLE
feat: Modifying the usage validity period

### PR DESCRIPTION
**Pull request by NitishAroraKPMG**
## Description

Adding the usage validity period in the netex file

## Testing instructions

Create the netex file or the export of the product made with fareDay end

### CODE REVIEW

File: repos/exporter/lib/handler.ts [hunk 1]
```diff
@@ -77,6 +78,7 @@ export const handler: Handler<ExportLambdaBody> = async ({ paths, noc, exportPre
             const fullProducts = ticketWithIds.products.map((product) => ({
                 ...product,
                 productName: product.productName ? product.productName : `${passengerTypeValue} ${ticketWithIds.type}`,
+                productEndTime: 'productValidity' in product && product.productValidity === 'fareDayEnd' ? fareDayEnd: undefined,
```
Comment (RobCattonKPMG) [inline repos/exporter/lib/handler.ts:81]: I dont think we need this

File: repos/fdbt-types/matchingJsonTypes.ts [hunk 2]
```diff
@@ -269,6 +269,7 @@ export interface ProductDetails extends BaseProduct {
     productPrice: string;
     productDuration: string;
     productValidity: string;
+    productEndTime?: string;
```
Comment (RobCattonKPMG) [inline repos/fdbt-types/matchingJsonTypes.ts:272]: I dont think we need this

File: repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts [hunk 3]
```diff
@@ -1052,7 +1052,11 @@ export const getPeriodConditionsElement = (
                             ? 'endOfFareDay'
                             : 'standardDuration',
                 },
-                ActivationMeans: { $t: 'noneRequired' },
+                ActivationMeans: product.productValidity === 'fareDayEnd' ? undefined : { $t: 'noneRequired' },
+                EndTime:
+                    product.productValidity === 'fareDayEnd' && 'fareDayEnd' in userPeriodTicket
```
Comment (RobCattonKPMG) [inline repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:1057]: If the productValidity is endOfCalenderDay, I think we should be setting EndTime to 2359

File: repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts [hunk 4]
```diff
@@ -1052,7 +1052,11 @@ export const getPeriodConditionsElement = (
                             ? 'endOfFareDay'
```
Comment (RobCattonKPMG) [inline repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:1052]: it looks like if `product.productValidity === 'endOfCalendarDay' || product.productValidity === 'fareDayEnd'`

then we are setting UsageEnd to endOfFareDay - which doesnt seem right

Comment (RobCattonKPMG) [reply repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:1052]: @NitishAroraKPMG could you investigate which values are allowed in this field? I think for endOfCalendarDay we need to be setting it to something different.

I think try setting the value to endOfCalendarDay and see if the netex allows it. And if it doesnt, it should hopefully tell us which values are allowed.

File: repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts [hunk 5]
```diff
@@ -1052,7 +1053,14 @@ export const getPeriodConditionsElement = (
...
                             : 'standardDuration',
                 },
-                ActivationMeans: { $t: 'noneRequired' },
+                EndTime:
+                    product.productValidity === 'fareDayEnd' &&
```
[the remark this answers is not in the document]
Comment (NitishAroraKPMG) [reply repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:1059]: it requires all the checks and tried after removing the check and does not working

[the remark this answers is not in the document]
Comment (NitishAroraKPMG) [reply repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:1059]: image

[the remark this answers is not in the document]
Comment (RobCattonKPMG) [reply repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts:1059]: okay, we shouldnt really need to do it but the types dont seem to be working properly

**Pull request merged**